Feature/nav bar

interim17 opened the pull request:
Problem
=======
Closes #372

Solution
========
Updated 12/9/25:
- NavBar exists as a component query with a markdown file that is not currently in the admin/config but can be added
- added `NavBarDropdown` component with a bunch of little handlers for shaping the data and getting the items props with CSS classnames for the antd component to consume
- NavBar dropdowns are populated and styled for Catalogs, Protocols, and Collections
-  added `/static/pdf` dir and new section in `gatsby-config.js` using `gatsby-source-filesystem` to make those pdfs accessible as files
- uploaded protocol pdfs from the old cell catalog site

Possible TODOs:
- styling could be tweaked on the `Collections` menu items its not pixel perfect yet.

* New feature (non-breaking change which adds functionality)

<img width="451" height="203" alt="Screenshot 2025-12-09 at 10 08 31 AM" src="https://github.com/user-attachments/assets/d95ce851-c9c4-4ed4-8457-79aafb3927b6" />
<img width="421" height="205" alt="Screenshot 2025-12-09 at 10 08 26 AM" src="https://github.com/user-attachments/assets/4308263f-a8d0-4e4e-93b4-249041d7b409" />
<img width="485" height="483" alt="Screenshot 2025-12-09 at 10 08 21 AM" src="https://github.com/user-attachments/assets/44619115-c244-4322-abfe-1ed91e3c7a8d" />
<img width="1254" height="304" alt="Screenshot 2025-12-09 at 10 08 13 AM" src="https://github.com/user-attachments/assets/00a433b4-a741-4b49-8bd3-e46700cd49b6" />

meganrm commented on src/style/navbar.module.css:
```diff
@@ -0,0 +1,37 @@
+.page-header {
```
nit: but we usually call the highest level class just `container`. I'm guessing this is just what we called it in cfe? Alternatively `nav-bar` would be more consistent with the component name

meganrm commented:
the hover state on the links out to correl end up having a little dot on the end, because the underline doesn't underline the link out icon but does underline the parentheses. A simple solution is to just move the linkout icon out of the parens because trying to have the underline just stop at the words seems like a pain

meganrm commented on src/component-queries/NavBar.tsx:
```diff
@@ -0,0 +1,107 @@
+                        }
+                        label="Catalogs"
+                        items={catalogs}
+                    />
```
looking more closely at the design, since so much of this is already hardcoded it seems like having this single component that has to do so much formatting is overly general. I would map over catalogs here and send them in as already formatted menu items (that don't open a new tab)

interim17 commented:
@meganrm Thanks for all the feedback!

Did a big re-org and a number of small fixes to bugs and styling.

Also sent Travis a couple questions about the catalog items.

Made one change for my sanity that I think you might hate and added an anchorType field to the markdown to make it simple to know at a glance if something is an internal/external link or a pdf download. It's just much easier for me mentally and makes downstream functions more readable.

Changes:
- NavBarDropdown directory exists with a `constants.ts` for the catalog component stuff, and a `formatDropDownMenuItems` file to handle formatting, which is called at the `NavBar` level
(Question, should I move NavBar into this folder? I left it in component queries because that what it is... I could put the directory there?)
- `PdfAnchor` and `LinkOut` components with prop to control new tab behavior
- made href field required to reduce checks
- added styling to triggers when dropdowns are open
- bumped icon outside the parens on collection links

meganrm commented on src/components/LinkOut.tsx:
```diff
@@ -0,0 +1,17 @@
+    newTab?: boolean;
+}
+
+export const LinkOut = ({label, href, newTab}: LinkOutProps) => {
```
nit: since this is called LinkOut, and you have an optional prop, I would think the default behavior is to have it go out. 

Also, right now the right hand links are not linking out, and I'm pretty certain they're supposed to be because they have the external icon (sorry if I put my comment about the left hand side not linking out in the wrong line and you changed it because of my comment). It can also be confusing to have an optional boolean because you don't know you're making a choice by not sending a prop in 
 
my suggestions: 
1. call this LinkOut, and have an optional `samePage` disable the default `target=_blank` behavior 
2. have this just called Link, and make a required prop that's an enum of either `SamePage` `NewTab` so you have to explicitly choose

meganrm commented on src/components/NavBarDropdown/formatDropDownMenuItems.tsx:
```diff
@@ -0,0 +1,71 @@
+    }
+
+    if (item.anchorType === NavBarAnchorType.Internal) {
+        return <LinkOut label={item.label} href={item.href}/>;
```
this is where link out looks confusing, i didn't notice the `newTab=true` below so it seemed weird to have Internal => LinkOut

meganrm commented on src/components/NavBarDropdown/formatDropDownMenuItems.tsx:
```diff
@@ -0,0 +1,71 @@
+    }
+
+    if (COLLECTION_COMPONENTS_MAP[item.label]) {
+        return <LinkOut label={COLLECTION_COMPONENTS_MAP[item.label]} href={item.href} />;
```
this needs `newTab={true}`

rugeli reviewed (approved):
clicked around the nav bar and everything worked as expected! 
One small thing I noticed on my end: I don’t see the parenthesis text in that menu, but it doesn’t seem to affect functionality. All good otherwise!
<img width="348" height="192" alt="Screenshot 2025-12-10 at 3 59 09 PM" src="https://github.com/user-attachments/assets/5413db1d-fad4-4949-932c-5acbce58f483" />

interim17 commented:
> clicked around the nav bar and everything worked as expected! One small thing I noticed on my end: I don’t see the parenthesis text in that menu, but it doesn’t seem to affect functionality. All good otherwise! <img alt="Screenshot 2025-12-10 at 3 59 09 PM" width="348" height="192" src="https://private-user-images.githubusercontent.com/91452427/525109790-5413db1d-fad4-4949-932c-5acbce58f483.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjU0NzI2MDUsIm5iZiI6MTc2NTQ3MjMwNSwicGF0aCI6Ii85MTQ1MjQyNy81MjUxMDk3OTAtNTQxM2RiMWQtZmFkNC00OTQ5LTkzMmMtNWFjYmNlNThmNDgzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTEyMTElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUxMjExVDE2NTgyNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTdjYjQ2MjljNjFjNmU5YjU2ZDQ2MDlhNGRjYTRhMmY3N2NiNGEzOTcyZTM3MzE1YmIzOGYxYzYwZTYxZDFkMTkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.UeEenppjjHT0WSjGCxG3BSI0sTl2AR93IX3WW8jSdU0">

I talked to Travis about this and we tweaked this design on the fly!
